java/client/src/org/openqa/selenium/firefox/GeckoDriverService.java

```diff
@@ -186,7 +192,8 @@ protected GeckoDriverService createDriverService(File exe, int port,
                                                      ImmutableList<String> args,
                                                      ImmutableMap<String, String> environment) {
       try {
-        GeckoDriverService service = new GeckoDriverService(exe, port, args, environment);
+        Duration timeout = (firefoxBinary == null) ? Duration.ofSeconds(45) : Duration.ofMillis(firefoxBinary.getTimeout());
```

Comment:
Why not calculate the timeout when someone calls `usingFirefoxBinary` on the builder?